test/e2e/e2e_test.go

```diff
@@ -258,6 +259,115 @@ float: 123.123
 	})
 }
 
+func TestSchema(t *testing.T) {
+	t.Run("scalar and map defaults", func(t *testing.T) {
+		flags := yttFlags{
+			{"--enable-experiment-schema": ""},
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema/config.yml", "../../examples/schema/schema.yml"}, "", flags, nil)
+		expectedOutput := `nothing: null
+string: a string
+bool: false
+int: 0
+float: 0.1
+any: anything
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("array defaults", func(t *testing.T) {
+		flags := yttFlags{
+			{"--enable-experiment-schema": ""},
+		}
+		actualOutput := runYtt(t, testInputFiles{"../../examples/schema-arrays/config.yml", "../../examples/schema-arrays/schema.yml"}, "", flags, nil)
+		expectedOutput := `jobs: []
+`
+
+		require.Equal(t, expectedOutput, actualOutput)
+	})
+
+	t.Run("--data-value-yaml flag", func(t *testing.T) {
```

Comment:
I'm thinking... we start losing returns in testing that multiple types are checked. (or am I missing something?)

I thought I might see a test here that shows that through all the different ways that Data Values are set (i.e. the full complement of `--data-value...` flags) all override schema defaults.

**Suggestion:** combine this test and the following and include the other `--data-value...` flags for a single test that puts the whole set of mechanisms through their paces.

Reply:
I modified `TestSchema()` to test all the different data values flags, similar to the `TestDataValues()` test works. This seems like the right thing to do since we aren't testing all these cases in [schema_test.go](https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/cmd/template/schema_test.go#L132-L175), and since in this examples [README.md](https://github.com/vmware-tanzu/carvel-ytt/pull/414/files#diff-d8bb9e365c5466d96d492fa351191026e428d1fda2502ce04c3d19073fc54efa) suggests using data values flags we have confidence that the example will work.